lib/pipeline/webpackProcess.js

```diff
@@ -82,27 +97,83 @@ class WebpackProcess extends ProcessWrapper {
             options: {
-              presets: ['babel-preset-es2016', 'babel-preset-es2017', 'babel-preset-react'].map(require.resolve),
-              plugins: ["babel-plugin-webpack-aliases"].map(require.resolve),
+              presets: [
+                [
+                  "@babel/preset-env", {
+                    modules: false,
+                    targets: {
+                      browsers: ["last 1 version", "not dead", "> 0.2%"]
+                    }
+                  }
+                ],
+                "@babel/preset-react"
+              ].map(pkg => {
+                if (Array.isArray(pkg)) {
+                  let _pkg = pkg[0];
+                  pkg[0] = require.resolve(_pkg);
+                  return pkg;
+                } else {
+                  return require.resolve(pkg);
+                }
+              }),
+              plugins: [
+                "@babel/plugin-transform-runtime",
+                "babel-plugin-webpack-aliases"
+              ].map(require.resolve),
               compact: false
             }
           }
         ]
       };
-    }
 
-    webpack(webpackOptions).run((err, stats) => {
-      if (err) {
-        console.error(err);
-      }
-      if (!detectErrors) {
-        return callback();
+      let dappBabelrc = path.join(realCwd, '.babelrc');
+      if (fs.existsSync(dappBabelrc)) {
+        webpackOptions.module.rules[3].options.extends = dappBabelrc;
       }
+    }
 
-      if (stats.hasErrors()) {
-        return callback(stats.toJson().errors.join("\n"));
-      }
-      callback();
+    webpack(webpackOptions).run((err, stats) => {
+      async.waterfall([
+        function checkStatsError(next) {
+          if (err) {
+            console.error(err);
+            return next(err);
+          }
+          if (!detectErrors) {
+            return next();
+          }
+          if (stats.hasErrors()) {
+            return next(
+              stats.toJson(webpackOptions.stats).errors.join("\n")
+            );
+          }
+          next();
+        }//,
+        // function writeStatsReport(next) {
+        //   if (detectErrors) {
+        //     self._log('info', 'writing file '+ ('.embark/stats.report').bold.dim);
+        //   }
+        //   fs.writeFile(
+        //     path.join(fs.dappPath('.embark'), 'stats.report'),
+        //     stats.toString(webpackOptions.stats),
+        //     next
+        //   );
+        // },
+        // function writeStatsJSON(next) {
+        //   if (detectErrors) {
+        //     self._log('info','writing file '+ ('.embark/stats.json').bold.dim);
+        //   }
+        //   fs.writeFile(
+        //     path.join(fs.dappPath('.embark'), 'stats.json'),
+        //     JSON.stringify(stats.toJson(webpackOptions.stats)),
+        //     next
+        //   );
+        // }
+        // note: to visualize the stats info in a browser, do...
+        // `npx webpack-bundle-analyzer <dapp_dir>/.embark/stats.json`
```

Comment:
Maybe put that with a debug flag that can be set later?

Reply:
Agreed. And/or a dapp author could activate them in a custom config. Or maybe both should be possible.